tools/integration_tests/readonly/list_objects_test.go

```diff
@@ -39,11 +39,11 @@ func TestListObjectsInBucket(t *testing.T) {
 		t.Errorf("The number of objects in the current directory doesn't match.")
 	}
 
-	if obj[0].Name() != DirectoryNameInTestBucket && obj[0].IsDir() != true {
+	if obj[0].Name() != DirectoryNameInTestBucket || obj[0].IsDir() != true {
```

Comment:
For Reviewer: if any of the condition fails, it will return error

Reply:
if this is a fix for list tests then please raise a separate PR for it.

Reply:
okay